### DISCUSSION TITLE
Contribute question process discussion

### DISCUSSION THREAD

**Discussion by AliceCodeZhang** [ECNU/OpenQuest#2] [Ideas]
1. 每个Question 需要包含的内容， 确保每个Question 能够被测评系统导入，我们需要包含以下几个关键字段

- 题目类型（必填） 
- 题干描述（必填） 
- 标签（7 大类 必填） 
- 知识点（必填） 
- 分值（必填） 
- 选项1（必填） | 分值？ | 
- 选项2（必填） | 分值？ | 
- 选项3 | 分值？ | 
- 选项4 | 分值？| 

2.  每个Question 在repo的提交流程

-     需求：需要至少两位reviewer, 多重验证之后能够合入
-     能够后期很好的被测评系统读取、更新和导入

> - **建议方式1** 
>     - 通过commit message 和GitHub review process 来管理每次的提交
> 
> ![image](https://github.com/ECNU/OpenQuest/assets/31360564/76bcc165-2cb5-4d06-8d8f-1fd2f4251c57)
> 
```
>    优点： 可以借助软件项目分支管理所有的题目，并为未来不同的需求可以通过分支来管理
>    - 所有提交进入主分支
>    - 主分支：管理所有的合入题目
>    - 分分支：可以拉取部分题目作为特定评测题目或者特定目的某次评测
>    - 同时题库拉取、更新的时候比较简单，可以通过action 来触发

>    缺点： 提交者需要熟悉代码提交方式
>    - 提交需要熟悉git workflow ，如何第一次正确的提交需要给出详细的文档帮助，给提交者仅需要约定最简单的方式，比如提交到main分支可能会减少操作的复杂度

```

- 建议方式2， 通过Issue的template 模板来提交题目，然后通过标签体系来进行review, 然后再通过自动化方式来获取review通过的题目，提交到repo

example:
![image](https://github.com/ECNU/OpenQuest/assets/31360564/b35b2f96-4691-4084-8720-4fb4b6c48d09)

```
>    优点： 提交者较为简单

>    缺点： 需要完善提交之后所有的标签体系和review体系以及自动化流程
>    - 需要构建新的issue +review +discussion 机制
>    - 需要将合格的issue（提交的question) 自动合入repo，可能需要调研下如何自动化
```

**Comment by andyhuang18** [comment]
题目如果需要除文本以外的描述信息（比如图片、音频、视频等）的话，是否可以在关键字段里增加一条选填内容：附件信息？

**Reply by AliceCodeZhang** [reply]
这个是个问题，如果是测评平台，短期第一版本可能只支持纯文本输入，但是后期确实需要支持题干和答案的多元化，目前在题目提交方面我们要想想咋弄比较合适

**Comment by andyhuang18** [comment]
我支持第一种方式，用分支管理更简洁

**Reply by AliceCodeZhang** [reply]
嗯嗯，第一种方式就是对提交者有一定的要求。 第二种的方法咱们需要找到一种可以把issue 自动合并到主干的方法，这样后面再使用主、分支整体管理项目，就是这个自动化方法需要找找有没有解决方案

**Comment by AliceCodeZhang** [comment]
1. 目前暂时确定用issue方式来提问
![image](https://github.com/ECNU/OpenQuest/assets/31360564/8c835a9c-31b5-4755-a5c1-6a6f53ba344b)



4. 一个新的问题最后被合并的案例

![image](https://github.com/ECNU/OpenQuest/assets/31360564/3f1c177b-1e94-4c1a-957a-c93b280c3168)
